Compiler/BackEnd/Tearing.mo

```diff
@@ -2911,8 +2911,15 @@ algorithm
   if debug then execStat("Tearing.ModifiedCellierHeuristic_3 - 2"); end if;
 
   // 2. Determine the variables in causEq
-  msel := Array.select(mIn,causEq);
-  potentialTVars := List.unique(List.flatten(arrayList(msel)));
+  //msel := Array.select(mIn,causEq);
+
+  potentialTVars := {};
+  for e in causEq loop
+    potentialTVars := listAppend(arrayGet(mIn, e), potentialTVars);
+  end for;
+  potentialTVars := List.unique(potentialTVars);
```

Comment:
In general I would be less worried by memory consumption (within limits) than by computation time. The availability of RAM is still increasing on PCs, while top speed of a single thread has basically stalled many years ago.